src/kbmod/search/common.h

```diff
@@ -53,9 +53,17 @@ struct Trajectory {
     // Whether the trajectory is valid. Used for on-GPU filtering.
     bool valid = true;
 
-    // Get pixel positions from a zero-shifted time.
-    float get_x_pos(float time) const { return x + time * vx; }
-    float get_y_pos(float time) const { return y + time * vy; }
+    // Get pixel positions from a zero-shifted time. Centered indicates whether
+    // the prediction starts from the center of the pixel (which it does in the search)
+    inline float get_x_pos(float time, bool centered = true) const {
+        return centered ? (x + time * vx + 0.5f) : (x + time * vx);
+    }
+    inline float get_y_pos(float time, bool centered = true) const {
+        return centered ? (y + time * vy + 0.5f) : (y + time * vy);
+    }
+
+    inline int get_x_index(float time) const { return (int)floor(get_x_pos(time, true)); }
+    inline int get_y_index(float time) const { return (int)floor(get_y_pos(time, true)); }
```

Comment:
I wonder if this deserves a `get_pos` that returns an `Index` or an `Point` and then snaps them to the respective grids. (For some reason this feels like it was done at one point but there are no traces of this in the code at all?)

On the GPU we have a separate `predict_index` anyhow and we should use our own inventions for consistency at least.